modules/ROOT/pages/domain-based-application-cookbook.adoc

```diff
@@ -0,0 +1,154 @@
+= Test Domain-Based Applications
+ifndef::env-site,env-github[]
+include::_attributes.adoc[]
+endif::[]
+
+MUnit 2.x allows you to write MUnit test suites for your Mule applications that share their resources with a Mule domain without the need for any extra configurations.
+
+== Base Mule Application
+
+Consider a Mule domain called `domain-config` that defines a global configuration for the HTTP listeners of all the Mule applications under it:
+
+[.ex]
+=====
+[discrete.view]
+===== Studio Visual Editor
+
+image::domain-munit-example.png[]
+
+
+[discrete.view]
+===== XML Editor
+
+.Mule domain
+[source,XML,linenums]
+----
+<domain:mule-domain ...>
+
+<http:listener-config name="HTTP_Listener_config_domain" basePath="/" >
+  <http:listener-connection host="0.0.0.0" port="8081" />
+</http:listener-config>
+
+
+</domain:mule-domain>
+----
+
+.Mule domain pom.xml file
+[source,XML,linenums]
+----
+<groupId>com.mycompany</groupId>
+<artifactId>domain-config</artifactId>
+<version>1.0.0-SNAPSHOT</version>
+<packaging>mule-domain</packaging>
+<name>domain-config</name>
+----
+=====
+
+And a basic Mule application that returns the path being pinged:
+
+[.ex]
+=====
+[discrete.view]
+===== Studio Visual Editor
+
+image::mule-app-domain-listener.png[]
+
+Note that this application does not define a HTTP Listener global configuration because it inherits it from its Mule domain (`HTTP_Listener_config_domain`).
+
+image::mule-app-domain-set-payload.png[]
+
+[discrete.view]
+===== XML Editor
+
+[source,XML,linenums]
+----
+<mule ...>
+
+  <flow name="application-aFlow">
+        <http:listener  config-ref="HTTP_Listener_config_domain" path="/api"/>
+        <set-payload value='#["You reached " ++ message.attributes.listenerPath]'/>
+  </flow>
+
+</mule>
+----
+
+Note that this application does not define an `http:listener-config` global element because it inherits it from its Mule domain (`HTTP_Listener_config_domain`). The application declares the domain dependency in its `pom.xml` file:
+
+[source,XML,linenums]
+----
+<dependencies>
+...
+  <dependency>
+    <groupId>com.mycompany</groupId>
+    <artifactId>domain-config</artifactId>
+    <version>1.0.0</version>
+    <classifier>mule-domain</classifier>
+    <scope>provided</scope>
+  </dependency>
+...
+</dependencies>
+----
+
+=====
+
+== Write The MUnit Test
+
+Because the Mule domain is a dependency of the Mule application, MUnit can start up the Mule domain to pick up the necessary global elements configuration for the application that it's testing.
```

Comment:
```suggestion
Because the Mule domain is a dependency of the Mule application, MUnit can start the Mule domain to retrieve the necessary global elements configuration for the application that it's testing.
```